### ISSUE TITLE
Can we get rid of `odd` and `even` classes in tables?

### ISSUE THREAD

**Issue by nisargjhaveri** [phpmyadmin/phpmyadmin#12070]
Can we get rid of the `odd` and `even` classes we use to style table rows? We seem to spend a lot of efforts handling that throughout the codebase.

Realized while looking at https://github.com/phpmyadmin/phpmyadmin/commit/d798809215c3147150dad12a14a3f049db3bbea4#diff-ec96a27edeea8db227299965812df402R176.

Alternative is to use `:nth-child([odd|even])` pseudo selector. Looks like it is well supported too. http://caniuse.com/#search=nth-child

Sorry for not using the template, the template is not really a good fit for this kind of issues.

**Comment by ibennetch** [comment]
I haven't looked through all the code on this, there might be an edge case or two where this wouldn't fit well — but short of anything like that, I think this is a great idea.

**Comment by nijel** [comment]
I think it's reasonable move. This comes from times this was not possible to achieve using CSS.

**Comment by madhuracj** [comment]
+1

**Comment by OlafvdSpek** [comment]
Is alternating row coloring still a good idea?
I think it went out of fashing a decade ago. ;)

**Comment by nijel** [comment]
I think it makes it easier to read, but this is something what can be done purely in CSS now.

**Comment by vosgus** [comment]
Hi, I have developed a fix for this issue. At the moment the fix is implemented in the pmahomme theme. Should I also implement it in the other themes (metro, original etc.)? Also I have only made changes to the CSS code since I'm not that familiar with the PHP codebase yet.

**Comment by nijel** [comment]
It would be great if you could port it to original theme (which is part of every release). The metro theme is separate, but fix for it is welcome as well.

**Comment by nijel** [comment]
Mostly fixed by #12287, but there are still some places left (this just just a bit filtered `git grep` output, it's probably not complete and might contain false positives):

```
db_datadict.php:    $odd_row = true;
db_datadict.php:        echo $odd_row ? 'odd' : 'even'; $odd_row = ! $odd_row;
db_export.php:$odd = true;
db_export.php:    $multi_values .= '<tr class="' . ($odd ? 'odd' : 'even') . ' marked">';
db_export.php:    $odd = ! $odd;
js/functions.js:            last_clicked_row = last_click_checked ? $table.find('tr.odd:not(.noclick), tr.even:not(.noclick)').index($tr) : -1;
js/functions.js:                $tr.parent().find('tr.odd:not(.noclick), tr.even:not(.noclick)')
js/functions.js:            var curr_row = $table.find('tr.odd:not(.noclick), tr.even:not(.noclick)').index($tr);
js/functions.js:            $tr.parent().find('tr.odd:not(.noclick), tr.even:not(.noclick)')
js/functions.js:    $(document).on('hover', 'tr.odd, tr.even',function (event) {
js/functions.js: * Also fixes the even/odd classes of the table rows at the end.
js/functions.js:        $(this).find('tr:odd')
js/functions.js:        .removeClass('even').addClass('odd')
js/functions.js:        .removeClass('odd').addClass('even');
js/functions.js:    elementList = $('tbody .odd,tbody .even');
js/functions.js:        if ($curr_row.hasClass('odd')) {
js/functions.js:            $curr_row.removeClass('odd').addClass('even');
js/functions.js:            $curr_row.removeClass('even').addClass('odd');
libraries/display_change_password.lib.php:        . '<tr class="odd">'
libraries/display_export.lib.php:            $class = 'odd';
libraries/display_export.lib.php:                $class = $class === 'odd' ? 'even' : 'odd';
libraries/gis/GISMultipolygon.php:            'fill-rule'    => 'evenodd',
libraries/gis/GISPolygon.php:            'fill-rule'    => 'evenodd',
libraries/navigation/Navigation.php:                    $odd = true;
libraries/navigation/Navigation.php:                        $html .= '<tr class="' . ($odd ? 'odd' : 'even') . '">';
libraries/navigation/Navigation.php:                        $odd = ! $odd;
libraries/rte/rte_list.lib.php:        $rowclass = ($count % 2 == 0) ? 'odd' : 'even';
libraries/rte/rte_list.lib.php: * @param string $rowclass Empty or one of ['even'|'odd']
libraries/rte/rte_list.lib.php: * @param string $rowclass Empty or one of ['even'|'odd']
libraries/rte/rte_list.lib.php: * @param string $rowclass Empty or one of ['even'|'odd']
libraries/rte/rte_routines.lib.php:                    $color_class = 'odd';
libraries/rte/rte_routines.lib.php:                        $color_class = ($color_class == 'odd') ? 'even' : 'odd';
libraries/rte/rte_routines.lib.php:        $rowclass = ($i % 2 == 0) ? 'even' : 'odd';
libraries/server_user_groups.lib.php:        $odd = true;
libraries/server_user_groups.lib.php:            $html_output .= '<tr class="' . ($odd ? 'odd' : 'even') . '">';
libraries/server_user_groups.lib.php:            $odd = ! $odd;
libraries/tracking.lib.php:    $style = 'odd';
libraries/tracking.lib.php:            $style = 'odd';
libraries/tracking.lib.php:    $style = 'odd';
libraries/tracking.lib.php:            $style = 'odd';
libraries/tracking.lib.php:    $style = 'odd';
libraries/tracking.lib.php:            $style = 'odd';
libraries/tracking.lib.php:    $style = 'odd';
libraries/tracking.lib.php:            $style = 'odd';
libraries/tracking.lib.php:    $style = 'odd';
libraries/tracking.lib.php: * @param string $style     odd|even
libraries/tracking.lib.php: * @return string $style        changed style (even|odd)
libraries/tracking.lib.php:            $style = 'odd';
libraries/tracking.lib.php:    $style = 'odd';
libraries/tracking.lib.php:            $style = 'odd';
templates/columns_definitions/partitions.phtml:        <?php $odd = true; ?>
templates/columns_definitions/partitions.phtml:            <tr class="<?= ($odd ? 'odd' : 'even'); ?>">
templates/columns_definitions/partitions.phtml:                <tr class="<?= ($odd ? 'odd' : 'even'); ?>">
templates/columns_definitions/partitions.phtml:            <?php $odd = !$odd; ?>
templates/database/structure/show_create_row.phtml:        <?php $odd = true; ?>
templates/database/structure/show_create_row.phtml:            <?php $row_class = ($odd) ? 'odd' : 'even'; ?>
templates/database/structure/show_create_row.phtml:            <?php $odd = ! $odd; ?>
templates/privileges/privileges_summary.phtml:                    <tr class="odd">
templates/privileges/privileges_summary.phtml:                    <?php $odd = true; ?>
templates/privileges/privileges_summary.phtml:                                            'odd'  => $odd,
templates/privileges/privileges_summary.phtml:                        <?php $odd = !$odd; ?>
templates/privileges/privileges_summary_row.phtml:<tr class="<?= ($odd ? 'odd' : 'even'); ?>">
templates/table/search/replace_preview.phtml:            <?php $odd = true; ?>
templates/table/search/replace_preview.phtml:                    <tr class="<?= ($odd ? 'odd' : 'even'); ?>">
templates/table/search/replace_preview.phtml:                    <?php $odd = ! $odd; ?>
templates/table/structure/display_partitions.phtml:                <?php $odd = true; ?>
templates/table/structure/display_partitions.phtml:                    <tr class="noclick <?= ($odd ? 'odd' : 'even') , ($hasSubPartitions ? ' marked' : '') ?>">
templates/table/structure/display_partitions.phtml:                                <tr class="noclick <?= $odd ? 'odd' : 'even' ?>">
templates/table/structure/display_partitions.phtml:                    <?php $odd = ! $odd; ?>
test/classes/DisplayResultsTest.php:                'odd',
test/classes/DisplayResultsTest.php:                '<td class="odd" class="center print_ignore"><input type'
test/classes/DisplayResultsTest.php:                'odd edit_row_anchor',
test/classes/DisplayResultsTest.php:                '<td class="odd edit_row_anchor center print_ignore"  >'
test/classes/DisplayResultsTest.php:                'odd',
test/classes/DisplayResultsTest.php:                '<td class="odd center print_ignore"  ><span class='
test/classes/DisplayResultsTest.php:                'odd',
test/classes/DisplayResultsTest.php:                '<td class="odd center print_ignore"  >' . "\n"
themes/original/css/printview.css:table tr.odd th,
themes/original/css/printview.css:table tr.odd td,
themes/original/css/printview.css:.odd {
themes/pmahomme/css/printview.css:table tr.odd th,
themes/pmahomme/css/printview.css:table tr.odd td,
themes/pmahomme/css/printview.css:.odd {
```
